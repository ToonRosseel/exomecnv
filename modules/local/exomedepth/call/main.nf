process EXOMEDEPTH_CALL {
    tag "$sample"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-exomedepth:1.1.16--r43hfb3cda0_3' :
        'biocontainers/r-exomedepth:1.1.16--r43hfb3cda0_3' }"

    input:
    tuple val(meta), path(countfile), val(sample), val(samples), val(families) // meta:id, chr, sam, fam, sample
    tuple val(meta2), path(exon_target) // meta.id=chrx/autosomal

    output:
    tuple val(meta), path("*.txt"), emit: cnvcall
    path "versions.yml", emit:versions

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.1.16'

    """
    ExomeDepth_cnv_calling.R \\
        $sample \\
        $countfile \\
        $exon_target \\
        $prefix \\
        $samples \\
        $families

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.1.16'
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ExomeDepth: ${VERSION}
        R: \$(Rscript --version | sed 's/R scripting front-end //g')
    END_VERSIONS
    """
}
