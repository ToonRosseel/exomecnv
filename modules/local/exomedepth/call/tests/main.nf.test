nextflow_process {

    name "Test Process EXOMEDEPTH_CALL"
    script "../main.nf"
    process "EXOMEDEPTH_CALL"

    test("homo_sapiens - bed, merged_counts") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test', chromosome: 'chrX' ],
                    file(params.merged_counts, checkIfExists:true),
                    "Sample1",
                    "Sample1,Sample2,Sample3,Sample4,Sample5",
                    "Fam1,Fam2,Fam3,Fam4,Fam5"
                ])
                input[1] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - bed, merged_counts - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test', chromosome: 'chrX' ],
                    file(params.merged_counts, checkIfExists:true),
                    "Sample1",
                    "Sample1,Sample2,Sample3,Sample4,Sample5",
                    "Fam1,Fam2,Fam3,Fam4,Fam5"
                ])
                input[1] = Channel.value([
                    [ id:'exon' ],
                    file(params.roi_auto)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
