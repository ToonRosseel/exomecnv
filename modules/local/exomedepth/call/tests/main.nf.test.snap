{
    "homo_sapiens - bed, merged_counts": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "Sample1_CNVs_ExomeDepth_autosomal.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,52fa9e72dbb0111366d891f166c93b95"
                ],
                "cnvcall": [
                    [
                        {
                            "id": "test"
                        },
                        "Sample1_CNVs_ExomeDepth_autosomal.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,52fa9e72dbb0111366d891f166c93b95"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T12:27:22.177728161"
    },
    "homo_sapiens - bed, merged_counts - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.autosomal.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,52fa9e72dbb0111366d891f166c93b95"
                ],
                "cnvcall": [
                    [
                        {
                            "id": "test"
                        },
                        "test.autosomal.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,52fa9e72dbb0111366d891f166c93b95"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T12:27:29.219396791"
    }
}